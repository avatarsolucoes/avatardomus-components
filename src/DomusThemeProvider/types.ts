export interface ThemeBgColor {
  color: string;
  background: string;
}

export interface IThemeData {
  primary: string;
  secundary: string;
  contrast: string;
  bgprimary: ThemeBgColor;
  bgsecundary: ThemeBgColor;
  bgcontrast: ThemeBgColor;
  charts: {
    primary: string;
    secundary: string;
    contrast: string;
    primaryContrast: string;
    secundaryContrast: string;
  };
  logo: {
    url: string;
  }
}

export interface IUseTheme {
  theme: IThemeData;
  getTheme: IThemeData;
  getDefaultTheme: IThemeData;
}
