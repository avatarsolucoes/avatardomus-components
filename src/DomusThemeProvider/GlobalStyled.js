import React, { useContext } from 'react';
import chroma from 'chroma-js';
import { createGlobalStyle } from 'styled-components';
import ThemeContext from './ThemeContext';

function prepareColorsName(theme, key) {
  const color = theme[key].color || theme[key];
  return chroma(color).alpha(1);
}

const Global = createGlobalStyle`
  @keyframes fadeinMain{
	  0% { opacity:0; overflow: hidden; }
	  100% { opacity:1; overflow: hidden; }
  }

  body {
    background-color: ${({ theme }) => theme.bgprimary.background};
    color: ${({ theme }) => prepareColorsName(theme, 'bgprimary')};
    min-height: 100%;
    overflow-x: hidden;
    a { color: ${({ theme }) => prepareColorsName(theme, 'bgprimary')} };
  }

  h1, h2 {
    font-size: 18px;
    margin: 0 auto;
    text-align: center;
    padding: 0;
    padding-bottom: 10px;
  }
  h3, h4 {
    margin: 0 auto;
    padding: 0 0 6px 0;
  }

  h3 { font-size: 20px; }
  h4 { font-size: 16px; }

  p {
    font-size: 16px;
    padding: 0 0 6px 0;
    margin: 0 auto;
  }
  p + h1, p + h2, p + h3, p + h4 { padding-top: 6px; }
  ul, ol {
    display: block;
    max-width: 100%;
    padding: 0 8px;
    margin: 0;
  }
  ul li, ol li {
    font-size: 16px;
    margin: 0 0 2px 8px;
    padding: 2px;
  }

  .footer {
    max-height: 44px;
    display: flex;
    padding: 3px;
    img {
      display: block;
      margin: 0 auto;
    }
  };

  #root {
    height: 100vh;
    margin: 0 auto;
    position: relative;
  }

  #layout {
    height: 100%;
    transition: all 0.2s linear;
    animation: fadeinMain 0.3s linear;
  }

  #header, .header-content {
    display: block;
    margin: 0 auto;
    padding: 0;
    position: relative;
    border: 0;
  }
  #header{
    padding: 8px 4px 26px 4px;
    background-repeat: repeat-x;
    background-position: bottom left;
    background-size: 390px 16px;
  }

  .header-content {
    width: 100%;
    max-width: 100%;
    display: flex;
    flex-flow: row nowrap;
    justify-content: center;
    align-items: center;
    align-content: center;
  }
  .header-content.between {
    justify-content: space-between;
    align-items: center;
  }

  .logo{
    display: block;
    margin:0 auto;
    max-height:44px;
  }

  .primary {
    color: ${({ theme }) => theme.primary}
  };

  .secundary {
    color: ${({ theme }) => theme.secundary}
  };

  .contrast {
    color: ${({ theme }) => theme.contrast}
  };

  .bgprimary {
    color: ${({ theme }) => theme.bgprimary.color};
    background: ${({ theme }) => theme.bgprimary.background};
  };

  .bgsecundary {
    color: ${({ theme }) => theme.bgsecundary.color};
    background: ${({ theme }) => theme.bgsecundary.background};
  };

  .bgcontrast {
    color: ${({ theme }) => theme.bgcontrast.color};
    background: ${({ theme }) => theme.bgcontrast.background};
  };

  @media screen and (min-width: 721px) {
	  h1, h2 { font-size: 26px; }
	  #header {
      padding: 14px;
      padding-bottom: 36px;
    }
	  .logo {
      max-height: 64px;
    }

	  .content {
      padding: 0 12px 12px 12px;
    }
  }

`;

export default function GlobalStyled() {
  const { theme } = useContext(ThemeContext);
  return <Global theme={theme} />;
}
