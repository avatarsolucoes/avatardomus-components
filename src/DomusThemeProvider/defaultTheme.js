import PropTypes from 'prop-types';
/**
 * @typedef {import('./types').IThemeData} IThemeData
 * @exports IThemeData
 */

/** @type {IThemeData} */
export const defaultTheme = {
  primary: '#122A54',
  secundary: '#0D7ECC',
  contrast: '#BB1806',
  bgprimary: { color: '#FFFFFF', background: '#122A54' },
  bgsecundary: { color: '#FFFFFF', background: '#0D7ECC' },
  bgcontrast: { color: '#FFFFFF', background: '#BB1806' },
  charts: {
    primary: '#8884d8',
    secundary: '#82ca9d',
    contrast: '#FF0000',
    primaryContrast: '#6f3b64',
    secundaryContrast: '#56b3CC',
  },
  logo: {
    url: '',
  },
  headerImageBottom: {
    image: '',
  },
};

export const shapeBgColor = {
  color: PropTypes.string,
  background: PropTypes.string,
};

export const shapeChartsColor = {
  primary: PropTypes.string,
  secundary: PropTypes.string,
  contrast: PropTypes.string,
  primaryContrast: PropTypes.string,
  secundaryContrast: PropTypes.string,
};

export const shapeLogo = {
  url: PropTypes.string,
};

export const shapeTheme = {
  primary: PropTypes.string,
  secundary: PropTypes.string,
  contrast: PropTypes.string,
  bgprimary: PropTypes.shape(shapeBgColor),
  bgsecundary: PropTypes.shape(shapeBgColor),
  bgcontrast: PropTypes.shape(shapeBgColor),
  charts: PropTypes.shape(shapeChartsColor),
  logo: PropTypes.shape(shapeLogo),
  headerImageBottom: PropTypes.shape(shapeLogo),
};
