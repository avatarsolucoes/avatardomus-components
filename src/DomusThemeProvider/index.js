export { default as DomusThemeProvider } from './DomusThemeProvider';
export { default as useTheme } from './useTheme';
export { default as GlobalStyled } from './GlobalStyled';
export { defaultTheme } from './defaultTheme';

/**
 * @typedef {import('./types').IThemeData} IThemeData
 * @typedef {import('./types').IUseTheme} IUseTheme
 *
 * @exports IThemeData
 * @exports IUseTheme
 */
