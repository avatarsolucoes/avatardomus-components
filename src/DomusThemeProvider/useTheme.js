import { useContext } from 'react';

import ThemeContext from './ThemeContext';

/**
 * @typedef {import('./types').IThemeData} IThemeData
 * @typedef {import('./types').IUseTheme} IUseTheme
 */

/**
 * @function useTheme
 * @returns {IUseTheme}
 */
export default function useTheme() {
  const { theme, getTheme, getDefaultTheme } = useContext(ThemeContext);
  return {
    theme,
    getTheme,
    getDefaultTheme,
  };
}
