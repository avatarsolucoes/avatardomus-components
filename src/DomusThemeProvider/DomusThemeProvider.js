import React from 'react';
import PropTypes from 'prop-types';
import ThemeContext from './ThemeContext';
import { defaultTheme, shapeTheme } from './defaultTheme';
import GlobalStyled from './GlobalStyled';

/**
 * @typedef {import('./types').IThemeData} IThemeData
 * @exports IThemeData
 * @exports defaultTheme
 */

export default function ThemeProvider(props) {
  const { children, theme } = props;

  /**
   * @function getTheme
   * @returns {IThemeData}
   */
  function getTheme() {
    return theme;
  }

  /**
   * @function getTheme
   * @returns {IThemeData}
   */
  function getDefaultTheme() {
    return defaultTheme;
  }
  return (
    <ThemeContext.Provider value={{ theme, getTheme, getDefaultTheme }}>
      <GlobalStyled />
      {children}
    </ThemeContext.Provider>
  );
}

ThemeProvider.defaultProps = {
  children: null,
  theme: defaultTheme,
};

ThemeProvider.propTypes = {
  children: PropTypes.any,
  // theme: PropTypes.any,
  theme: PropTypes.shape(shapeTheme),
};
