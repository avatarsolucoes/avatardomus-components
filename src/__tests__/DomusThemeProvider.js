import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import { DomusThemeProvider, useTheme } from '../DomusThemeProvider';

function RenderChildren1() {
  const { theme } = useTheme();
  return <div style={{ backgroundColor: theme.primary }} />;
}

function RenderChildren2() {
  const { getTheme } = useTheme();
  const theme = getTheme();
  return <div style={{ backgroundColor: theme.secundary }} />;
}

function RenderChildren3() {
  const { getDefaultTheme } = useTheme();
  const theme = getDefaultTheme();
  return <div style={{ backgroundColor: theme.contrast }} />;
}

function RenderDomusThemeProvider() {
  return (
    <DomusThemeProvider>
      <RenderChildren1 />
      <RenderChildren2 />
      <RenderChildren3 />
    </DomusThemeProvider>
  );
}

describe('DomusThemeProvider', () => {
  it(`is truthy Span`, () => {
    expect(Text).toBeTruthy();
  });

  it(`Test renderer DomusThemeProvider`, () => {
    const wrapper = {
      text: renderer.create(<RenderDomusThemeProvider />).toJSON(),
    };
    expect(wrapper).toMatchSnapshot();
  });

  it(`Test shallow DomusThemeProvider`, () => {
    const wrapper = shallow(<RenderDomusThemeProvider />);
    expect(wrapper).toMatchSnapshot();
  });

  it(`Test renderer Children`, () => {
    const wrapper = {
      a: renderer.create(<RenderChildren1 />).toJSON(),
      b: renderer.create(<RenderChildren2 />).toJSON(),
      c: renderer.create(<RenderChildren3 />).toJSON(),
    };
    expect(wrapper).toMatchSnapshot();
  });
});
