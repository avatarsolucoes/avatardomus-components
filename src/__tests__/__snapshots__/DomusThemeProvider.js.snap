// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DomusThemeProvider Test renderer Children 1`] = `
Object {
  "a": <div
    style={
      Object {
        "backgroundColor": "#122A54",
      }
    }
  />,
  "b": <div
    style={
      Object {
        "backgroundColor": "#0D7ECC",
      }
    }
  />,
  "c": <div
    style={
      Object {
        "backgroundColor": "#BB1806",
      }
    }
  />,
}
`;

exports[`DomusThemeProvider Test renderer DomusThemeProvider 1`] = `
Object {
  "text": Array [
    <div
      style={
        Object {
          "backgroundColor": "#122A54",
        }
      }
    />,
    <div
      style={
        Object {
          "backgroundColor": "#0D7ECC",
        }
      }
    />,
    <div
      style={
        Object {
          "backgroundColor": "#BB1806",
        }
      }
    />,
  ],
}
`;

exports[`DomusThemeProvider Test shallow DomusThemeProvider 1`] = `ShallowWrapper {}`;
